tests/test_salesforce_switch_rep_method_ft_incrmntl.py

```diff
@@ -0,0 +1,96 @@
+        catalog_entries = [ce for ce in found_catalogs if ce['tap_stream_id'] in expected_streams]
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+        streams_replication_methods = {stream: self.FULL_TABLE
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # Run a sync job using orchestrator
+        fulltbl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        fulltbl_sync_records = runner.get_records_from_target_output()
+
+        fulltbl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        #Switch the replication method from full table to Incremental
+        streams_replication_methods = {stream: self.INCREMENTAL
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # SYNC 2
+        incrmntl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        incrmntl_sync_records = runner.get_records_from_target_output()
+        incrmntl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # Test by stream
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # record counts
+                fulltbl_sync_count = fulltbl_sync_record_count.get(stream, 0)
+                incrmntl_sync_count = incrmntl_sync_record_count.get(stream, 0)
+                replication_key = list(replication_keys[stream])[0]
+
+                # Verify at least 1 record was replicated in the Incrmental sync
+                self.assertGreater(incrmntl_sync_count, 0,
+                                   msg="We are not fully testing bookmarking for {}".format(stream))
+                # data from record messages
+                primary_key = list(primary_keys[stream])[0]
+                fulltbl_sync_messages = [record['data'] for record in
+                                       fulltbl_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                fulltbl_primary_keys = {message[primary_key] for message in fulltbl_sync_messages}
+                incrmntl_sync_messages = [record['data'] for record in
+                                        incrmntl_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                incrmntl_primary_keys = {message[primary_key] for message in incrmntl_sync_messages}
+
+                #Verify all records are synced in the second sync
+                self.assertTrue(fulltbl_primary_keys.issubset(incrmntl_primary_keys))
+
+                #verify that the last message is not a activateversion message for incremental sync
+                self.assertNotEqual('activate_version', incrmntl_sync_records[stream]['messages'][-1]['action'])
```

Comment:
The activate_version message tells the loaders it is okay to go ahead and load data into the table. If we have sent all of the data, it is okay to start loading.  So although this message may not be necessary, it would have no negative consequences.

Also, If they take the appropriate action and fix it so that they do not send the activate_version as the first message for a new table version for incremental, waiting until all data is ready before deleting the previous full table records, Then it would be necessary/critical to have this message be sent.

So in the final version we should assert that the only the last message is an activate version, and as a workaround we should verify that there is an activate message being sent.

Reply:
Added a qa task in the DOD of the ticket to modify the assertions based on the final implementation.